src/main/java/calculator/domain/Parser.java

```diff
@@ -0,0 +1,77 @@
+package calculator.domain;
+
+import calculator.StringUtils;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Parser {
+
+    private static final String[] DEFAULT_DELIMITER = {",", ":"};
+    private static final String CUSTOM_DELIMIT_START_STRING = "//";
+    private static final int CUSTOM_DELIMITER_INDEX = 2;
+    private static final int CUSTOM_DELIMITER_LAST_INDEX = 5;
+
+    private String input;
+
+    public Parser(String input) {
+        this.input = input;
+    }
+
+    public List<Integer> getIntegers() {
+        if (StringUtils.isEmpty(input)) {
+            return Collections.singletonList(0);
+        }
+
+        if (isCustom()) {
+            String[] split = checkCustomDelimit().split(findCustomDelimiter());
+            return changeToIntegers(split);
+        }
+
+        String[] split = input.split(Arrays.toString(DEFAULT_DELIMITER));
+        return changeToIntegers(split);
+    }
+
+    private boolean isCustom() {
+        return input.startsWith(CUSTOM_DELIMIT_START_STRING);
+    }
+
+    private String checkCustomDelimit() {
+        if (!input.contains("\\n")) {
+            throw new IllegalArgumentException("커스텀구분자 정의는 //커스텀구분자\n 형식으로 입력해주세요.");
+        }
+        return input.substring(CUSTOM_DELIMITER_LAST_INDEX);
+    }
+
+    private String findCustomDelimiter() {
+        return String.valueOf(input.charAt(CUSTOM_DELIMITER_INDEX));
+    }
+
+    private List<Integer> changeToIntegers(String[] split) {
+
+        return Arrays.stream(split)
+                .map(this::changeToInteger)
+                .collect(Collectors.toList());
+    }
+
+    private Integer changeToInteger(String string) {
+        Integer integer;
+        try {
+            integer = Integer.parseInt(string);
```

Comment:
```suggestion
            integer = Integer.parseUnsignedInt(string);
```

기존에 제공하고 있는 API를 활용하는 것은 어떤가요.
내부적으로 음수일 경우에 Exception을 던지고 있습니다.

```java
            if (firstChar == '-') {
                throw new
                    NumberFormatException(String.format("Illegal leading minus sign " +
                                                       "on unsigned string %s.", s));
```

Reply:
저는 자료형을 메서드명 혹은 변수명에 사용하는 것은 지양하고 있습니다.

https://github.com/Yooii-Studios/Clean-Code/blob/master/Chapter%2002%20-%20의미%20있는%20이름.md#의미-있게-구분하라불용어-noise-word-를-쓰지-말자

Reply:
저는 정수의 의미로 쓰고 싶었는데요~ 자료형과 이름이 같은건 피해야겠네요.
number 로 변경해볼게요 !!